libraries/AP_Airspeed/AP_Airspeed_SST_ND.cpp

```diff
@@ -0,0 +1,292 @@
+
+uint8_t config_setting[2] = {0x54, 0x00}; // notch filter disabled, bw limit set to 50Hz-> 148Hz odr with auto select, wdg disabled, pressure range set to 0b100
+uint8_t sst_config_setting[2] = {0x0A, 0x07}; //bw limit set to 50Hz -> 155.35Hz, pressure range set to 0b010
+
+const float *p_current_range;
+
+// probe for a sensor
+bool AP_Airspeed_SST_ND::probe(uint8_t bus, uint8_t address)
+{
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
+        return false;
+    }
+    WITH_SEMAPHORE(_dev->get_semaphore());
+
+    _dev->set_retries(20);
+    uint8_t reading[14];
+
+    if(!_dev->read(reading, 14)){
+        return false;
+    }
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Found bus %u addr 0x%02x", _dev->bus_num(), _dev->get_bus_address());
+    return matchModel(&reading[6]);
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_SST_ND::init()
+{
+    static const uint8_t addresses[] = { ND_I2C_ADDR1, ND_I2C_ADDR2 };
+    if (bus_is_confgured()) {
+        for (uint8_t addr : addresses) {
+            if (probe(get_bus(), addr)) {
+                goto found_sensor;
+            }
+        }
+    } else {
+        FOREACH_I2C_EXTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+        FOREACH_I2C_INTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+    }
+
+    //GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "SST_ND[%u]: no sensor found", get_instance());
+    return false;
+
+found_sensor:
+    _dev->set_device_type(uint8_t(DevType::SST_ND));
+    set_bus_id(_dev->get_bus_id());
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"SST_ND[%u]: Found bus %u addr 0x%02x", get_instance(), _dev->bus_num(), _dev->get_bus_address());
+
+    // send default configuration
+    WITH_SEMAPHORE(_dev->get_semaphore());
+    //_dev->transfer(config_setting, 2, nullptr,0);
+
+    switch(_dev_model){
+        case DevModel::SST_ND:
+            _available_ranges = ARRAY_SIZE(vn131cm_range);
+            p_current_range = vn131cm_range;
+            _range_setting = 7;
+            break;
+        case DevModel::ND210:
+            _available_ranges = ARRAY_SIZE(nd210_range);
+            p_current_range = nd210_range;
+            _range_setting = 6;
+            break;
+        case DevModel::ND005D:
+            _available_ranges = ARRAY_SIZE(nd005d_range);
+            p_current_range = nd005d_range;
+            _range_setting = 6;
+            break;
+        case DevModel::ND160:
+            _available_ranges = ARRAY_SIZE(nd160_range);
+            p_current_range = nd160_range;
+            _range_setting = 7;
+            break;
+        case DevModel::ND130:
+            _available_ranges = ARRAY_SIZE(nd130_range);
+            p_current_range = nd130_range;
```

Comment:
We need to do a bit of work on the names here.

Leading underscores vs not, for example.  Our current preference is no leading underscores on members.

Perhaps `available_ranges` -> `range_max` and `p_current_range` -> `range` and `_range_setting` -> `range_ofs`